hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/ddl/TestCreateTable.scala

```diff
@@ -1555,4 +1555,34 @@ class TestCreateTable extends HoodieSparkSqlTestBase {
       )("Failed to create catalog table in metastore")
     }
   }
+
+  test("Test Create Table with Same Value for Partition and Precombine") {
```

Comment:
Could we add more tests covering different branches / cases introduced in `partitionColumnsToRead`?